src/Realm.C

```diff
@@ -2561,6 +2561,10 @@ Realm::set_current_coordinates(stk::mesh::Part* targetPart)
   VectorFieldType* displacement = meta_data().get_field<VectorFieldType>(
     stk::topology::NODE_RANK, "mesh_displacement");
 
+  modelCoords->sync_to_host();
+  currentCoords->sync_to_host();
```

Comment:
I think this is correct but overly aggressive.
Since currentCoords is going to be over-written, a sync-to-host seems unnecessary.
But the modify_on_host for it down below definitely seems necessary. Because that will signal any subsequent sync_to_device calls that the data does indeed need to be sync'd.

Reply:
We just had a successful run with this commit so I am going to merge as is.   We can get less aggressive later. :)